### PR TITLE
Add mole reactor science pages

### LINKED ISSUE #244: Mole-based reactors are undocumented
The governing equations for the mole-based reactor types (`MoleReactor`, `ConstPressureMoleReactor`, `IdealGasMoleReactor`, and `IdealGasConstPressureMoleReactor`) need to be documented here: https://cantera.org/science/reactors/reactors.html#reactor-types-and-governing-equations.

@anthony-walker, can you help with this?

### PR DESCRIPTION
**Pull request by anthony-walker**
The mole based reactors are currently undocumented in the science pages. This adds science pages for them. It is my intention to more content in regards to preconditioning. This closes #244.

### CODE REVIEW

File: pages/science/reactors/constpresmolereactor.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      More information on the Constant Pressure Mole Reactor class can be found `here.
+      <{{% ct_docs doxygen/html/d5/d7d/classCantera_1_1ConstPressureMoleReactor.html %}}>`__
+
+Constant Pressure Mole Reactor
+******************************
+
+For this reactor model, the pressure is held constant. The energy equation is
+defined by the total enthalpy.
+
+A Constant Pressure Reactor is defined by the two state variables:
```
Comment (speth) [inline pages/science/reactors/constpresmolereactor.rst:24]:
```suggestion
A Constant Pressure Mole Reactor is defined by the two state variables:
```

File: pages/science/reactors/constpresmolereactor.rst [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+The moles of each species in the reactor's contents changes as a result of flow through
+the reactor's inlets and outlets, and production of homogeneous gas phase species and reactions on the reactor :py:class:`Wall`.
+The rate of moles of species :math:`k` generated through homogeneous phase
+reactions is :math:`V \dot{\omega}_k`, and the total rate at which moles of species
+:math:`k` is generated is:
+
+.. math::
+
+   \frac{dn_k}{dt} = V \dot{\omega}_k + \sum_{in} \dot{n}_{in} - \sum_{out} \dot{n}_{out} + \dot{n}_{wall}
+   \tag{2}
```
Comment (speth) [inline pages/science/reactors/constpresmolereactor.rst:63]: I think you're missing $k$ subscripts on all the $\dot{n}$ terms. Also, I don't think there's a formula for $n_{wall}$ given anywhere. On the parent page for reactors, the walls section provides only the related formula for $m_{k,wall}$.

File: pages/science/reactors/idealgasconstpresmolereactor.rst [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+   \frac{dH}{dt} = \frac{dT}{dt}\sum_k n_k \bar{c_{p,k}} + \sum \bar{h}_k \dot{n}_k
+
+After some manipulations yields an equation for the
+temperature:
+
+.. math::
+
+   \frac{dT}{dt} = \frac{\dot{Q} - \sum \bar{h}_k \dot{n}_k}{\sum_k n_k \bar{c}_{p,k} }
+  \tag{1}
+
```
Comment (speth) [inline pages/science/reactors/idealgasconstpresmolereactor.rst:44]: For consistency with the RHS/LHS separation that we use, I would suggest keeping the $\sum_k n_k \bar{c}_{p,k}$ on the left hand side.

**Review by speth — CHANGES_REQUESTED** [review]
Thanks @anthony-walker. I think this mostly looks good. I have a few suggestions. Please note that most of these comments apply across multiple reactor variants.

**Comment by anthony-walker** [comment]
@speth Thank you for the quick turn around on review. I made the changes you suggested and I added some basic discussion and an example about preconditioning in the `CVODES` page.

**Review by speth — APPROVED** [review]
Thanks, @anthony-walker, this looks good to me.

**Pull request merged**